dapp-devs-updates (#177)
ethereum-optimism/docs: pages/dapp-developers/dapp-developer-overview.mdx

---
title: Dapp Developer Overview
lang: en-US
description: Learn about deploying contracts, cross-chain messaging, and tutorials to help you build applications for Optimism.
---

# Dapp Developer Overview

This document provides an overview of tools and resources to help you build applications for Optimism. It points you to guides on basic contract deployment, cross-chain messaging, and [Dapp developer tutorials](#dapp-developer-tutorials).
If you're looking for third-party tools that make building apps on OP Mainnet easier, check out the [Developer Tools](/tools/overview) section.

## Getting Started

> If you're brand new to OP Mainnet, we recommend starting with the guide on [deploying a basic contract](/getting-started-op-mainnet).

Reviewer: ```suggestion
If you're brand new to OP Mainnet, try starting with the guide on [deploying a basic contract](/getting-started-op-mainnet).
```

It'll get you familiar with the core steps required to get a contract deployed to the network.
Luckily, OP Mainnet is [EVM equivalent](https://medium.com/ethereum-optimism/introducing-evm-equivalence-5c2021deb306), so it's 100% the same as deploying a contract to Ethereum.

You might also want to check out the [testing on OP Networks guide](testing) and the tutorial on [running a local development environment](/testing/dev-node) to help you feel totally confident in your OP Mainnet deployment.

## Cross-Chain Messaging

> Looking to explore cross-chain messaging? We've got detailed guides for that.

Reviewer: ```suggestion
Looking to explore cross-chain messaging? Here are some detailed guides for that.
```

> If you want to bridge a token from Ethereum to OP Mainnet (or vice versa!), you should check out our [Standard Token Bridge](bridging/standard-bridge).

Reviewer: ```suggestion
If you want to bridge a token from Ethereum to OP Mainnet (or vice versa!), you should check out the [Standard Token Bridge](bridging/standard-bridge).
```

The Standard Token Bridge makes the process of moving tokens between chains as easy as possible.

> If you're looking for something more advanced, we recommend reading through our guide on [sending data between L1 and L2](bridging/messaging).

Reviewer: ```suggestion
If you're looking for something more advanced, take a look at the guide on [sending data between L1 and L2](bridging/messaging).
```

Contracts on one chain can trigger contract functions on the other chain, it's pretty cool!

> We even dogfood the same infrastructure and use it under the hood of the Standard Token Bridge.

Reviewer: ```suggestion
OP Mainnet even uses this same message-passing infrastructure under the hood of the Standard Token Bridge.
```

| [Using OP Mainnet System Contracts](tutorials/system-contracts) | Learn how to work with OP Mainnet contracts directly from other contracts *and* how to work with contracts from the client side. | 🟢 Easy |
| [Bridging ETH with the Optimism SDK](tutorials/cross-dom-bridge-eth) | Learn how to use the Optimism SDK to transfer ETH between Layer 1 (Ethereum or Goerli) and Layer 2 (OP Mainnet or OP Goerli). | 🟢 Easy |
| [Bridging ERC-20 tokens with the Optimism SDK](tutorials/cross-dom-bridge-erc20) | Learn how to use the Optimism SDK to transfer ERC-20 tokens between Layer 1 (Ethereum or Goerli) and Layer 2 (OP Mainnet or OP Goerli). | 🟢 Easy |
| [Bridging your Standard ERC-20 token to OP Mainnet using the Standard Bridge](tutorials/standard-bridge-standard-token) | Learn how to bridge your standard ERC-20 token to OP Mainnet using the standard bridge. | 🟡 Moderate |
| [Bridging your Custom ERC-20 token to OP Mainnet using the Standard Bridge](tutorials/standard-bridge-custom-token) | Learn how to bridge your custom ERC-20 token to OP Mainnet using the standard bridge. | 🟡 Moderate |

> You can also [suggest a new tutorial](https://github.com/ethereum-optimism/docs/issues/new?assignees=&labels=tutorial%2Cdocumentation%2Ccommunity-request&projects=&template=suggest_tutorial.yaml&title=%5BTUTORIAL%5D+Add+PR+title) if you have something specific in mind. We'd love to grow our tutorials list for Dapp developers! 

Reviewer: ```suggestion
You can also [suggest a new tutorial](https://github.com/ethereum-optimism/docs/issues/new?assignees=&labels=tutorial%2Cdocumentation%2Ccommunity-request&projects=&template=suggest_tutorial.yaml&title=%5BTUTORIAL%5D+Add+PR+title) if you have something specific in mind. We'd love to grow this list! 
```

## Next Steps

If you still can't find the content you're looking for, there's a few options to get extra help. 

> *   You can [ask a question in our developer support forum](https://github.com/ethereum-optimism/developers/discussions).

Reviewer: ```suggestion
*   You can [ask a question in the Optimism developer support forum](https://github.com/ethereum-optimism/developers/discussions).
```
